include/live_model.h

```diff
@@ -220,6 +220,54 @@ namespace reinforcement_learning {
      */
     int report_outcome(const char* event_id, float outcome, api_status* status= nullptr);
 
+    /**
+     * @brief Report partial outcome of a decision based on an index.
+     *
+     * @param event_id  The unique event_id used when choosing an action should be presented here.  This is so that
+     *                  the action taken can be matched with feedback received.
+     * @param outcome Outcome as float.
+     * @param index Index of the partial outcome.
+     * @param status  Optional field with detailed string description if there is an error
+     * @return int Return error code.  This will also be returned in the api_status object
+     */
+    int report_outcome(const char* event_id, float outcome, int index, api_status* status= nullptr);
+
+    /**
+     * @brief Report partial outcome of a decision based on an index.
+     *
+     * @param event_id  The unique event_id used when choosing an action should be presented here.  This is so that
+     *                  the action taken can be matched with feedback received.
+     * @param outcome Outcome as float.
+     * @param index Index of the partial outcome.
+     * @param status  Optional field with detailed string description if there is an error
+     * @return int Return error code.  This will also be returned in the api_status object
+     */
+    int report_outcome(const char* event_id, float outcome, const char *index, api_status* status= nullptr);
```

Comment:
We maybe need to think about naming here. For multistep case it make sense to rename event_id (from envelope) to episode_id or session_id, and index to event_id (but in this case it brings extra confusion with non-multistep methods)

Reply:
Hmmm, yeah.
Actually I'm not that super excited with this API sprawl, what about introducing some sort of "struct outcome_id" and have a single method instead?

Reply:
Or couldn't we template this `report_outcome` method same as we template the `report_outcome_internal`?

Reply:
This is top level API call, so I am not sure if it is a good idea to template it.

Reply:
Regarding outcome_id structure - really not sure:
Although episode_id (join_id) + index are somehow logically represents outcome id, if we move to example builder in the future, these arguments should be structured in a way that is more aligned with schema: {join_id, {index, value}}.
So I would leave arguments structure plain for now.

Reply:
I prototyped an `outcome_id` struct and it felt like over-engineering.

Let's agree to revisit this once we're either adding one more parameter to report_outcome or another type to the existing pair. As long as the strings don't have any internal structure, I'm happy.

Reply:
sure sounds good to me